Tasks/AzurePowerShellV4/Tests/ChecksForPowerShellCore.ps1

```diff
@@ -0,0 +1,28 @@
+[CmdletBinding()]
+param()
+
+# Arrange.
+. $PSScriptRoot\..\..\..\Tests\lib\Initialize-Test.ps1
+$targetAzurePs = "4.1.0"
+Register-Mock Get-VstsInput { "FilePath" } -- -Name ScriptType -Require
+Register-Mock Get-VstsInput { "$PSScriptRoot/PerformsBasicFlow_TargetScript.ps1" } -- -Name ScriptPath
+Register-Mock Get-VstsInput { $targetAzurePs } -- -Name TargetAzurePs
+Register-Mock Get-VstsInput { 'arg1 arg2' } -- -Name ScriptArguments
+Register-Mock Get-VstsInput { "continue" } -- -Name errorActionPreference
+Register-Mock Get-VstsInput { $true } -- -Name FailOnStandardError
+Register-Mock Get-VstsInput { $true } -- -Name pwsh
+Register-Mock Get-VstsInput { ([System.IO.Path]::GetTempPath()) } -- -Name workingDirectory
+Register-Mock Update-PSModulePathForHostedAgent
+Register-Mock Get-Module
+Register-Mock Initialize-AzModule
+Register-Mock Get-VstsEndpoint { @{auth = @{ scheme = "ServicePrincipal" }} }
+Register-Mock Remove-EndpointSecrets
+Register-Mock Disconnect-AzureAndClearContext
+Register-Mock Assert-VstsPath
+Register-Mock Invoke-VstsTool { }
+
+# Act.
+$actual = & $PSScriptRoot\..\AzurePowerShell.ps1
+
+Assert-WasCalled Invoke-VstsTool -Times 1
+Assert-WasCalled Invoke-VstsTool -ArgumentsEvaluator {$args.count -eq 6 -and $args[3] -like '*pwsh.exe*'}
```

Comment:
Can you try this. $args | ConvertTo-Json) -like "*powershell.exe*"

Reply:
done